### ISSUE TITLE
Enhance SelectInput to return the selected choice onChange to set state on Wizard form review step

### ISSUE THREAD

**Issue by wattry** [marmelab/react-admin#9850]
**Is your feature request related to a problem? Please describe.**
When using the select input we sometimes need to make a decision or add optionText as a display value. 
Specifically when trying to set a display value for the review step of a Wizard form. 
Autocomplete passes back the choice onChange, which allows us to populate a context provider with the corresponding display value for the optionValue, often times it is an id which is meaningless  to the user.

**Describe the solution you'd like**
We'd like for the SelectInput and SelectArrayInput, preferably all choice inputs, to follow the same behavior of the autocomplete input by passing back the selected choice onChange events.

**Additional context**

Here is the documentation
[AutoCompleteInput](https://marmelab.com/react-admin/AutocompleteInput.html#onchange)

[AutocompleteArrayInput](https://marmelab.com/react-admin/AutocompleteArrayInput.html#onchange)

**Comment by slax57** [comment]
Sounds like a fair idea to me!
Feel free to open a PR if you'd like to give a try implementing this (remember to target the `next` branch if you do :slightly_smiling_face: ).

**Comment by slax57** [comment]
@wattry Yes, you can request for a development to get prioritized via the [Professional Services](https://react-admin-ee.marmelab.com/#professional-services) offer. As an Enterprise subscriber you are eligible to a 50% discount.

**Comment by fzaninotto** [comment]
Adding a reference to #9245 (published in 4.14), where this was done for AutocompleteInput.

**Comment by wattry** [comment]
Caveat emptor.
